src/Sulu/Bundle/WebsiteBundle/EventListener/RedirectExceptionListener.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\EventListener;
+
+use Sulu\Bundle\WebsiteBundle\Locale\DefaultLocaleProviderInterface;
+use Sulu\Component\Localization\Localization;
+use Sulu\Component\Webspace\Analyzer\Attributes\RequestAttributes;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Sulu\Component\Webspace\Url\ReplacerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Matcher\RequestMatcherInterface;
+
+/**
+ * This event-listener redirect trailing slashes and ".html" and redirects to default locale for partial-matches.
+ */
+class RedirectExceptionListener
+{
+    const HTML = '.html';
+
+    /**
+     * @var RequestMatcherInterface
+     */
+    private $router;
+
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @var DefaultLocaleProviderInterface
+     */
+    private $defaultLocaleProvider;
+
+    /**
+     * @var ReplacerInterface
+     */
+    private $urlReplacer;
+
+    /**
+     * @param RequestMatcherInterface $router
+     * @param RequestAnalyzerInterface $requestAnalyzer
+     * @param DefaultLocaleProviderInterface $defaultLocaleProvider
+     * @param ReplacerInterface $urlReplacer
+     */
+    public function __construct(
+        RequestMatcherInterface $router,
+        RequestAnalyzerInterface $requestAnalyzer,
+        DefaultLocaleProviderInterface $defaultLocaleProvider,
+        ReplacerInterface $urlReplacer
+    ) {
+        $this->router = $router;
+        $this->requestAnalyzer = $requestAnalyzer;
+        $this->defaultLocaleProvider = $defaultLocaleProvider;
+        $this->urlReplacer = $urlReplacer;
+    }
+
+    /**
+     * Redirect trailing slashes or ".html".
+     *
+     * @param GetResponseForExceptionEvent $event
+     */
+    public function redirectTrailingSlashOrHtml(GetResponseForExceptionEvent $event)
+    {
+        $request = $event->getRequest();
+        $route = $this->getRoute($request);
```

Comment:
The method is almost doing nothing, maybe it is also worth thinking about inlining it here?